### PR TITLE
[Genesis] Collapse all validator configs into a single struct and add support for separate accounts

### PR DESCRIPTION
**Pull request by movekevin**
### Description
Currently we pass a bunch of vectors to initialize validator accounts. This PR collapses this all into a single struct and also adds separate operator/voter account support in genesis.

### Test Plan
Existing unit and e2e tests

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2937)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/genesis.move [hunk 1]
```diff
@@ -110,48 +122,49 @@ module aptos_framework::genesis {
...
-        staking_distribution: vector<u64>,
-    ) {
-        let num_owners = vector::length(&owners);
-        let num_validator_network_addresses = vector::length(&validator_network_addresses);
-        let num_full_node_network_addresses = vector::length(&full_node_network_addresses);
-        assert!(num_validator_network_addresses == num_full_node_network_addresses, 0);
-        let num_staking = vector::length(&staking_distribution);
-        assert!(num_full_node_network_addresses == num_staking, 0);
+    fun create_initialize_validators(aptos_framework_account: signer, validators_serialized: vector<u8>) {
+        let validators = from_bytes<vector<ValidatorConfiguration>>(validators_serialized);
```
Comment (gregnazario) [inline aptos-move/framework/aptos-framework/sources/genesis.move:126]: Interesting, didn't know you could serialize types like this

File: aptos-move/vm-genesis/src/lib.rs [hunk 2]
```diff
@@ -308,35 +309,14 @@ fn create_and_initialize_validators(
...
             MoveValue::Signer(CORE_CODE_ADDRESS),
-            MoveValue::Vector(owners),
-            MoveValue::Vector(consensus_pubkeys),
-            MoveValue::Vector(proof_of_possession),
-            MoveValue::Vector(validator_network_addresses),
-            MoveValue::Vector(full_node_network_addresses),
-            MoveValue::Vector(staking_distribution),
+            MoveValue::vector_u8(bcs::to_bytes(validators).expect("Validators can be serialized")),
         ]),
     );
```
Comment (davidiw) [inline aptos-move/vm-genesis/src/lib.rs:321]: Is this right?  it kinda seems like you should just be placing a bcs inside a bcs -- maybe...

File: aptos-move/framework/aptos-framework/sources/genesis.move [hunk 3]
```diff
@@ -110,48 +122,49 @@ module aptos_framework::genesis {
...
-        full_node_network_addresses: vector<vector<u8>>,
-        staking_distribution: vector<u64>,
-    ) {
-        let num_owners = vector::length(&owners);
-        let num_validator_network_addresses = vector::length(&validator_network_addresses);
-        let num_full_node_network_addresses = vector::length(&full_node_network_addresses);
-        assert!(num_validator_network_addresses == num_full_node_network_addresses, 0);
-        let num_staking = vector::length(&staking_distribution);
-        assert!(num_full_node_network_addresses == num_staking, 0);
+    fun create_initialize_validators(aptos_framework_account: signer, validators_serialized: vector<u8>) {
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/genesis.move:125]: oh jeez dude.... just define 

vec<ValidatorConfiguration> and play with it until you get it right...

**Review by gregnazario — APPROVED** [review]

**Review by davidiw — APPROVED** [review]

File: aptos-move/vm-genesis/src/lib.rs (hunk 2 above)
Comment (movekevin) [reply aptos-move/vm-genesis/src/lib.rs:321]: It works and is so much cleaner 🤷‍♂️

File: aptos-move/framework/aptos-framework/sources/genesis.move (hunk 3 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/genesis.move:125]: lolll but why? The struct is so much cleaner :)

Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/genesis.move:125]: how is this cleaner??

you can still pass in `validators: vec<ValidatorConfiguration>`

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/genesis.move:125]: Is that possible without much more work to convert from the Rust struct to MoveValue::Struct fields?

**Pull request merged**